scripts/west_commands/runners/core.py

```diff
@@ -558,7 +558,7 @@ def add_parser(cls, parser):
             parser.add_argument('-i', '--dev-id', help=argparse.SUPPRESS)
 
         if caps.flash_addr:
-            parser.add_argument('--dt-flash', default='n', choices=_YN_CHOICES,
+            parser.add_argument('--dt-flash', default=False, choices=_YN_CHOICES,
```

Comment:
This doesn't make sense to me. `False` is not in `_YN_CHOICES`. The point of `action=_DTFlashAction` was to do the conversion to bool based on the value. If that's not working, the solution is to fix the action, not change the default to a value that isn't in `choices`.

Reply:
The action is not run if there is no value passed to the argument, only if the value is passed through the command line. So without argument the final value is 'n', but if we pass `--dt-flash n`, then `_DTFlashAction` is called, and we correctly get `False`.

Reply:
LGTM. If you wish, add a comment to help the next person who comes across this like:

`Note that _DTFlashAction converts the choice into a boolean`